kube-controllers/pkg/controllers/node/hostendpoints.go

```diff
@@ -346,32 +471,50 @@ func (c *autoHostEndpointController) getAutoHostendpointExpectedIPs(node *libapi
 	return expectedIPs
 }
 
-// generateAutoHostendpointFromNode returns the expected auto hostendpoint to be
-// created from the given node.
-func (c *autoHostEndpointController) generateAutoHostendpointFromNode(node *libapi.Node) *api.HostEndpoint {
-	hepLabels := make(map[string]string, len(node.Labels)+1)
+// generateAutoHostEndpoint returns a HostEndpoint created based on the specific parameters
+func (c *autoHostEndpointController) generateAutoHostEndpoint(node *libapi.Node, templateLabels map[string]string, hepName string, expectedIPs []string, interfaceName string) *api.HostEndpoint {
+	hepLabels := make(map[string]string)
 	for k, v := range node.Labels {
 		hepLabels[k] = v
 	}
+	for k, v := range templateLabels {
+		if _, ok := hepLabels[k]; ok {
+			logrus.Warnf("overwriting duplicate label %s", k)
```

Comment:
```suggestion
                        f := logrus.Fields{"key": k, "nodeVal": hepLabels[k], "userVal": v}
			logrus.WithFields(f).Warn("overwriting label from underlying Node resource")
```

We should almost never be putting variables into the log text - using Fields makes the logs more consistent and adds structure, which is valuable when ingesting logs into a log processor.